### ISSUE TITLE
No way to disable mse_loss broadcasting warning

### ISSUE THREAD

**Issue by EtienneDesticourt** [pytorch/pytorch#26551]
## 🐛 Bug

Hi, 

torch.nn.functional.mse_loss always throws a warning when using it with two tensors of different shapes.
There are legitimate reasons for wanting to use differently shaped tensor and taking advantage of the standard broadcasting behaviour of pytorch, so there needs to be a way to disable that warning.

## To Reproduce

```
import torch
import torch.nn.functional as F

a = torch.randn(5, 4)
b = torch.randn(5, 1)

F.mse_loss(a, b)
```

## Expected behavior

There should be an optional parameter to disable the warning, like `warn_broadcasting=True`, `error_broadcasting=False`.
As discussed here: https://github.com/pytorch/pytorch/issues/16045#issuecomment-476266780

## Additional context

There are legitimate cases for wanting to calculate the MSE between differently shaped tensors.
For example I need to calculate the difference between each sample and a subset of other samples. So I need to calculate the MSE between tensors shaped like:
(number of samples, dimension of a sample, 1) and (number of samples, dimension of a sample, number of samples in the subset).
I understand that safeguards need to be put in-place to avoid misleading people, as discussed in the original issue (https://github.com/pytorch/pytorch/issues/16045) but broadcasting is a Pytorch staple and users shouldn't have to re-implement basic functions like the MSE to make use of its full power.

Cheers

**Comment by colesbury** [comment]
Yes, patches welcome. cc @gchanan who mentioned this in the original issue

**Comment by colesbury** [comment]
FYI, as a workaround you can use:

a) `F.mse_loss(a, b.expand_as(a))` in your example
b) The [`warning`](https://docs.python.org/3/library/warnings.html) module to suppress warnings
